### PR TITLE
Shadow module documentation

### PR DESCRIPTION
**Pull request by orangejenny**
https://github.com/dimagi/commcare-hq/pull/9923#issuecomment-172477108

@snopoke

### CODE REVIEW

**Comment by orangejenny** [comment]
no test coverage => killed build

**Comment by orangejenny** [comment]
reminder to myself to update https://confluence.dimagi.com/display/ccinternal/Shadow+Modules to link to this doc once this PR is merged

**Comment by czue** [comment]
:sunny: for the :busts_in_silhouette:!

File: docs/advanced_app_features.rst [hunk 1]
```diff
@@ -119,3 +119,114 @@ which the child module forms do not. So we need to add it in:
...
+        </text>
+        <command id="m1-f0"/>
+    </menu>
+    <menu root="m2" id="m1.m2">                                                                                                     <text>
+            <locale id="modules.m1"/>
+        </text>                                                                                                                     <command id="m1-f0"/>
+    </menu>
+    <menu id="m2">                                                                                                                  <text>
+            <locale id="modules.m2"/>
+        </text>                                                                                                                     <command id="m2-f0"/>
```
Comment (snopoke) [inline docs/advanced_app_features.rst]: I think this should be `m0-f0`. Any reason why the order is different (just makes it harder to compare)?

Comment (snopoke) [inline docs/advanced_app_features.rst]: Ah, think I'm wrong about the command ID here but then is the command for module `m1.m2` correct? Should it not be `m1.m2-f0`?

File: docs/advanced_app_features.rst [hunk 2]
```diff
@@ -119,3 +119,114 @@ which the child module forms do not. So we need to add it in:
...
+The shadow module inherits from its source:
+
+- case type
+- commands (which forms the module leads to)
+- end of form behavior
+
+Limitations
+~~~~~~~~~~~
+
+A shadow module neither **be** a parent module nor **have** a parent module
```
Comment (snopoke) [inline docs/advanced_app_features.rst]: A shadow module **can** neither

**Comment by snopoke** [comment]
Great docs thanks @orangejenny

File: docs/advanced_app_features.rst (hunk 1 above)
Comment (orangejenny) [inline docs/advanced_app_features.rst]: No, the command ids don't get the shadow module's id appended, only the menu ids do.

File: docs/advanced_app_features.rst (hunk 2 above)
Comment (orangejenny) [inline docs/advanced_app_features.rst]: fixed

File: docs/advanced_app_features.rst (hunk 1 above)
Comment (snopoke) [inline docs/advanced_app_features.rst]: OK so in the case where a ShadowModule shadows a 'parent' module it does not duplicate the child module's entries (just the child module's menu) but it does duplicate the parent module's entries, correct?

Comment (orangejenny) [inline docs/advanced_app_features.rst]: Yep, it does not duplicate the child module's entries, only the parent module's entries.

**Pull request merged**